### DISCUSSION TITLE
max7219

### DISCUSSION THREAD

**Discussion by GKTRAINS** [community/community#135516] [Programming Help]
### Select Topic Area

Question

### Body

need help on connecting 8 max 7219 together for scrolling a message.
using arduino nano or arduino uno.
frustrated
thx
gk

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @GKTRAINS ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by erencanbas** [comment]
You can use the **LedControl** or **MD_MAX72XX** and **MD_Parola** libraries, which are powerful and flexible for handling multiple MAX7219 modules.

Install MD_MAX72XX and MD_Parola.
```

#include <MD_Parola.h>
#include <MD_MAX72XX.h>
#include <SPI.h>


#define MAX_DEVICES 8
#define DATA_PIN    11
#define CLK_PIN     13
#define CS_PIN      10

MD_Parola P = MD_Parola(CS_PIN, MAX_DEVICES);

void setup() {
  P.begin();
  P.setIntensity(5); 
  P.displayText("Hello World!", PA_CENTER, 100, 50, PA_SCROLL_LEFT, PA_SCROLL_LEFT);
}

void loop() {
  if (P.displayAnimate()) {
    P.displayReset();
  }
}
```

**Comment by GKTRAINS** [comment]
Eren

Thx - but no work

Only 1 and 2 scroll.....

3-4-5-6-7-8 stay lit  up while 1 and 2 scroll????

thx
gary


________________________________

**Comment by erencanbas** [comment]
**Zones:** The setZone() function is used to ensure that the entire range of devices (0 to 7) is included in a single zone. This allows the scrolling effect to span across all 8 devices.

This should allow your message to scroll across all 8 devices without leaving any of them statically lit up.

```
#include <MD_Parola.h>
#include <MD_MAX72XX.h>
#include <SPI.h>

#define MAX_DEVICES 8
#define DATA_PIN    11
#define CLK_PIN     13
#define CS_PIN      10

MD_Parola P = MD_Parola(CS_PIN, MAX_DEVICES);

void setup() {
  P.begin();
  P.setIntensity(5);


  P.setZone(0, 0, MAX_DEVICES - 1);  // Zone 0 covers all devices


  P.displayText("Hello World!", PA_CENTER, 100, 50, PA_SCROLL_LEFT, PA_SCROLL_LEFT);
}

void loop() {
  if (P.displayAnimate()) {
    P.displayReset();
  }
}

```
